### DISCUSSION TITLE
Help with error in Swift 5

### DISCUSSION THREAD

**Discussion by hbenditsky** [community/community#64637] [Programming Help]
### Body

my repository is My-Books-Read.I am new to programming.  I have created a small database program in Swift with the help of CodeAI that. has an error I don'r understand or know how to fix. The file is appended below with error to be found noted at bottom of page 3 and top of page 4 of code. Would appreciate help.Thanks
[My.Books.Read.pdf](https://github.com/community/community/files/12411412/My.Books.Read.pdf)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by jeria93** [comment]
what most likely causes the error is:

There’s a literal "ERROR:" "Trailing closure passed to parameter of type 'FormStyleConfiguration' that does not accept a closure" in your Form block. This string isn’t commented out, so Swift treats it as an unexpected argument, resulting in a trailing closure error.

Your saveItem function is missing a closing brace, which causes the cancel function to be nested incorrectly. This misplacement further confuses the compiler about where your closures begin and end.

It should look somethings like this:

private func saveItem() {
    do {
        try item.managedObjectContext?.save()
        presentationMode.wrappedValue.dismiss()
    } catch {
        print("Error saving item: \(error.localizedDescription)")
    }
}

private func cancel() {
    presentationMode.wrappedValue.dismiss()
}
